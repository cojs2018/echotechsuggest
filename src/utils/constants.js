export const bookmarksAPI = 'https://5zcbdoz0h3.execute-api.eu-west-2.amazonaws.com/dev';