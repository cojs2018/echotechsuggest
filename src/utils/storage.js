export async function createBookmark(bookmarkUrl) {
    return {
        status: 200,
        message: 'Bookmark created successfully!'
    }
}

export async function listBookmarks() {
    return [
        {
            bookmarkId: "0",
            articleName: "article0 - articlearchive",
            path: 'http://www.articlearchive.com/article0',
            url: new URL('http://www.articlearchive.com/article0'),
            createdAt: (new Date()).toUTCString(),
            tags: []
        },
        {
            bookmarkId: "1",
            articleName: "article1 - articlearchive",
            path: 'http://www.articlearchive.com/article1',
            url: new URL('http://www.articlearchive.com/article1'),
            createdAt: (new Date()).toUTCString(),
            tags: []
        },
        {
            bookmarkId: "2",
            articleName: "article2 - articlearchive",
            path: 'http://www.articlearchive.com/article2',
            url: new URL('http://www.articlearchive.com/article2'),
            createdAt: (new Date()).toUTCString(),
            tags: [ "tag 1" ]
        }
    ];
}

export async function deleteBookmark(bookmarkName) {
    return {
        status: 200,
        message: 'Bookmark deleted successfully!'
    }
}