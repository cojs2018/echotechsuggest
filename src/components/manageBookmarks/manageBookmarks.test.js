jest.mock('../../utils/storage');

import React from 'react';
import { deleteBookmark, listBookmarks } from '../../utils/storage';
import { mount } from 'enzyme';
import render from 'react-test-renderer';
import ManageBookmarks from './manageBookmarks'

const getMockDate = () => {
    const randomTime = Math.floor(Math.random() * Date.now());
    const mockDate = new Date(randomTime);
    return mockDate.toUTCString();
};

const mockBookmarkUrls = [
    {
        id: "0",
        articleName: "article0 - articlearchive",
        path: 'http://www.articlearchive.com/article0',
        url: new URL('http://www.articlearchive.com/article0'),
        createdAt: getMockDate(),
    },
    {
        id: "1",
        articleName: "article1 - articlearchive",
        path: 'http://www.articlearchive.com/article1',
        url: new URL('http://www.articlearchive.com/article1'),
        createdAt: getMockDate(),
    },
    {
        id: "2",
        articleName: "article2 - articlearchive",
        path: 'http://www.articlearchive.com/article2',
        url: new URL('http://www.articlearchive.com/article2'),
        createdAt: getMockDate(),
        tags: [ "tag 1" ]
    }
]

listBookmarks.mockReturnValue(mockBookmarkUrls);

// React statas and effects
let columns = [];
let rows = [];
let selected = [];
let alertMessage = {
    status: 'none',
    message: ''
}

const setColumns = jest.fn(async (newColumnSet) => columns = newColumnSet);
const setRows = jest.fn(async (newRowSet) => rows = newRowSet);
const setSelected = jest.fn(async (selectedRows) => selected = selectedRows);
const setAlertMessage = jest.fn(async (newMessage) => alertMessage = newMessage);

beforeEach(() => {
    React.useState = jest.fn()
        .mockReturnValueOnce([columns, setColumns])
        .mockReturnValueOnce([rows, setRows])
        .mockReturnValueOnce([selected, setSelected])
        .mockReturnValueOnce([alertMessage, setAlertMessage]);
    
    React.useEffect = jest.fn()
        .mockImplementationOnce(effectFunction => effectFunction());
})

describe('Manage Bookmark', () => {
    it('renders correctly', () => {
        const tree = render
            .create(<ManageBookmarks />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    })
});