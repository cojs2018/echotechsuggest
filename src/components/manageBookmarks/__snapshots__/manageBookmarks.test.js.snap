// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Manage Bookmark renders correctly 1`] = `
<View
  style={
    Object {
      "width": "97%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#6200ee",
        "elevation": 4,
        "flexDirection": "row",
        "height": 56,
        "paddingHorizontal": 4,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 3,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 4,
      }
    }
  >
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "flex": 1,
            "paddingHorizontal": 12,
          },
          Array [
            false,
            false,
            undefined,
          ],
        ]
      }
    >
      <Text
        accessibilityRole="header"
        accessibilityTraits="header"
        accessible={true}
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
            },
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "fontSize": 17,
              },
              undefined,
            ],
          ]
        }
      >
        Manage Bookmarks
      </Text>
    </View>
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      focusable={true}
      hitSlop={
        Object {
          "bottom": 6,
          "left": 6,
          "right": 6,
          "top": 6,
        }
      }
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "margin": 6,
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 18,
              "height": 36,
              "width": 36,
            },
            Object {
              "opacity": 0.32,
            },
            undefined,
          ],
        ]
      }
      testID="delete"
    >
      <View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
              Object {
                "height": 24,
                "width": 24,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "rotate": "0deg",
                  },
                ],
              }
            }
          >
            <Text
              accessibilityElementsHidden={true}
              allowFontScaling={false}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontSize": 24,
                  },
                  Array [
                    Object {
                      "lineHeight": 24,
                      "transform": Array [
                        Object {
                          "scaleX": 1,
                        },
                      ],
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ],
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              󰆴
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": undefined,
        }
      }
      accessible={true}
      focusable={true}
      hitSlop={
        Object {
          "bottom": 6,
          "left": 6,
          "right": 6,
          "top": 6,
        }
      }
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "margin": 6,
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 18,
              "height": 36,
              "width": 36,
            },
            undefined,
            undefined,
          ],
        ]
      }
      testID="refresh"
    >
      <View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
              Object {
                "height": 24,
                "width": 24,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "rotate": "0deg",
                  },
                ],
              }
            }
          >
            <Text
              accessibilityElementsHidden={true}
              allowFontScaling={false}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontSize": 24,
                  },
                  Array [
                    Object {
                      "lineHeight": 24,
                      "transform": Array [
                        Object {
                          "scaleX": 1,
                        },
                      ],
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ],
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              󰑐
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View>
    NONE: 
  </View>
  <View>
    Data Table
  </View>
</View>
`;
